warning: variable `unused` is assigned to, but never used
  --> $DIR/is-long.rs:28:20
   |
28 |     my_opt is Some(unused); //~ WARN variable `unused` is assigned to, but never used
   |                    ^^^^^^
   |
note: lint level defined here
  --> $DIR/is-long.rs:11:9
   |
11 | #![warn(unused)]
   |         ^^^^^^
   = note: #[warn(unused_variables)] implied by #[warn(unused)]
   = note: consider using `_unused` instead

warning: value assigned to `unused` is never read
  --> $DIR/is-long.rs:28:5
   |
28 |     my_opt is Some(unused); //~ WARN variable `unused` is assigned to, but never used
   |     ^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/is-long.rs:11:9
   |
11 | #![warn(unused)]
   |         ^^^^^^
   = note: #[warn(unused_assignments)] implied by #[warn(unused)]

error[E0381]: use of possibly uninitialized variable: `x`
  --> $DIR/is-long.rs:38:9
   |
38 |     let _x = x; //~ ERROR use of possibly uninitialized variable: `x`
   |         ^^ use of possibly uninitialized `x`

error[E0381]: use of possibly uninitialized variable: `y`
  --> $DIR/is-long.rs:39:9
   |
39 |     let _y = y; //~ ERROR use of possibly uninitialized variable: `y`
   |         ^^ use of possibly uninitialized `y`

error[E0381]: use of possibly uninitialized variable: `z`
  --> $DIR/is-long.rs:40:9
   |
40 |     let _z = z; //~ ERROR use of possibly uninitialized variable: `z`
   |         ^^ use of possibly uninitialized `z`

error: aborting due to 3 previous errors

