warning: variable `a` is assigned to, but never used
  --> $DIR/is.rs:34:28
   |
34 |     let _ = my_opt is Some(a); //~ WARN variable `a` is assigned to, but never used
   |                            ^
   |
note: lint level defined here
  --> $DIR/is.rs:13:9
   |
13 | #![warn(unused)]
   |         ^^^^^^
   = note: #[warn(unused_variables)] implied by #[warn(unused)]
   = note: consider using `_a` instead

warning: value assigned to `a` is never read
  --> $DIR/is.rs:34:13
   |
34 |     let _ = my_opt is Some(a); //~ WARN variable `a` is assigned to, but never used
   |             ^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/is.rs:13:9
   |
13 | #![warn(unused)]
   |         ^^^^^^
   = note: #[warn(unused_assignments)] implied by #[warn(unused)]

