error[E0425]: cannot find value `x` in this scope
  --> $DIR/is.rs:14:8
   |
14 |        x > 10 { //~ ERROR cannot find value `x` in this scope
   |        ^ not found in this scope

error[E0425]: cannot find value `x` in this scope
  --> $DIR/is.rs:15:26
   |
15 |         println!("{:?}", x); //~ ERROR cannot find value `x` in this scope
   |                          ^ not found in this scope

error[E0425]: cannot find value `x` in this scope
  --> $DIR/is.rs:20:11
   |
20 |           x > 10 { //~ ERROR cannot find value `x` in this scope
   |           ^ not found in this scope

error[E0425]: cannot find value `x` in this scope
  --> $DIR/is.rs:21:26
   |
21 |         println!("{:?}", x); //~ ERROR cannot find value `x` in this scope
   |                          ^ not found in this scope

error: aborting due to 4 previous errors

